### DISCUSSION TITLE
Option to disable GET of /files endpoint

### DISCUSSION THREAD

**Discussion by metalmarco** [directus/directus#20908] [Feature Requests]
### Summary

For security reasons, I think it would be a good idea to have an option to disable the GET request to the the `/files` endpoint without specifying an explicit ID.

In this way only GET requests directed to a specific file would be allowed, but GET requests that could allow a user to list ALL the files in Directus storage will be disallowed-

### Basic Example

GET `<directus>/files `--> 401
GET `<directus>/files/<uuid>` --> 200

### Motivation

Making sure that files stored in Directus storage are safe from unauthorized access it's very important.
The best solution from a security point of view, which is currently possibile in Directus, is to define granular READ permission on the `directus_files` collection so that they are based on every possible relation the specific file have within the intere datamodel.

But for non critical applications, the "protection" granted by the UUID could be enough. A user will have to bruteforce UUIDs to get to a specific file.
However, if there is no way to block access to the GET requests addressed to the `/files` endpoint, any user can LIST all the files available in the storage.

### Detailed Design

This could be addressed by an ENV variable which could disable the GET requests to a collection when no `/<id>` is specified

### Must Have:
- Configuration variable that allow to disable get requests to the `/files` endpoint without specificing an `/<id>` in the requests

### Should Have:
- Ability to extend the block to any collection in the database

### Could Have:
- Could be implemented in the Permissions App, without using ENV variables

### Won't Have:
- Shoudn't be enabled by default, for regression compatiblity

### Drawbacks

No drawbacks, as long as the feature is disabled by default and well-documented.
If enabled, the Files view on the Directus App could be unusable.

### Alternatives

The alternative is to set READ permission based on every relation the directus_files collection have on the database. This could be a very long activity when files are related to dozens of tables.

### Adoption Strategy

The solution would bring an extra layer of security.
I don't think it's breaking change but documentation should point out that, if the option is enabled, could affect existing applications.

**Comment by br41nslug** [comment]
Be aware that disabling the `/files` GET endpoint (for authenticated users) will also require disabling the files module and file/image interfaces in the App which both depend on listing files using that endpoint. Interfaces will keep working but you cannot select new items from the list.

**Comment by rbochenski** [comment]
@metalmarco did you find solution for this?

**Reply by metalmarco** [reply]
Unfortunately not

**Reply by rbochenski** [reply]
i found it, i have modified this extension https://github.com/br41nslug/directus-extension-block-endpoints

```
	init("app.before", ({ app }: any) => {
		paths.forEach(p => {
			const [path, pattern] = p.split('::');
			app.get(path, (_req: any, res: any, next: any) => {
				if (!_req.headers['host'].startsWith('public-api')) {
					return next();
				}

				if (pattern) {
					const regex = new RegExp(pattern);
					if (!regex.test(_req.path)) {
						return next();
					}
				}
				res.set("Content-Type", type);
				res.status(status);
				res.send(body);
			});
		});
	});
```

env variables:
```
BLOCKED_ENDPOINTS_ENABLED=true
BLOCKED_ENDPOINTS_PATHS=/files::/files[/]?$,/server/info

```

**Reply by rbochenski** [reply]
i have normal api for application protected by firewall, public-api is accessible without FW

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
